pkg/util/hostname/common.go

```diff
@@ -103,6 +104,11 @@ func isOSHostnameUsable(ctx context.Context) (osHostnameUsable bool) {
 	}
 
 	// Check UTS namespace from docker
+	// TODO: Revisit when we introduce support for Windows privileged containers
+	if runtime.GOOS == "windows" {
+		return false
+	}
+
```

Comment:
As discussed, I think this change the behavior on Windows and will produce a different hostname when running on k8s with `net=host` option.

Also we landed a major rewrite of the hostname detection for 7.39. Would it be possible to postpone this change to 7.40 in order to have less change in this for the next release ? The hostname detection is critical.

Reply:
That's the whole point, there's no `net=host` on Windows, so the call is useless, it will always return Unknown